File: src/main/java/com/jss/osiris/modules/quotation/service/MailRedirectionTypeService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.MailRedirectionType;

public interface MailRedirectionTypeService {
    public List<MailRedirectionType> getMailRedirectionTypes();

    public MailRedirectionType getMailRedirectionType(Integer id);
}

File: client-myjss/src/app/modules/quotation/services/mail.redirection.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from '../../main/services/appRest.service';
import { MailRedirectionType } from '../model/MailRedirectionType';

@Injectable({
  providedIn: 'root'
})
export class MailRedirectionTypeService extends AppRestService<MailRedirectionType> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getMailRedirectionTypes() {
    return this.getListCached(new HttpParams(), "mail-redirection-types");
  }
}
